website/server/controllers/api-v3/members.js

```diff
@@ -402,6 +362,109 @@ function _getMembersForItem (type) {
   };
 }
 
+async function _getMemberTasksFromChallenge (challengeId, memberId, user, res) {
+  const member = await User.findById(memberId).select(`${nameFields} challenges`).exec();
+  if (!member) throw new NotFound(res.t('userWithIDNotFound', { userId: memberId }));
+
+  const challenge = await Challenge.findById(challengeId).exec();
+  if (!challenge) throw new NotFound(res.t('challengeNotFound'));
+
+  // optionalMembership is set to true because even if you're
+  // not member of the group you may be able to access the challenge
+  // for example if you've been booted from it, are the leader or a site admin
+  const group = await Group.getGroup({
+    user, groupId: challenge.group, fields: '_id type privacy', optionalMembership: true,
+  });
+  if (!group || !challenge.canView(user, group)) throw new NotFound(res.t('challengeNotFound'));
+  if (!challenge.isMember(member)) throw new NotFound(res.t('challengeMemberNotFound'));
+
+  const challengeTasks = await Tasks.Task.find({
```

Comment:
I was thinking, since the number of results could be pretty high here we should probably use `.lean()` to make sure the server doesn't incur in performance issues

Reply:
The only problem I see is that it uses custom `.toJson()` schema method which replaces `_id` with `id` and (probably?) makes some other changes to the data. I can create some util method like `modelToJson(options)` and use it inside of both [`schema.options.toJson.transform` function](https://github.com/HabitRPG/habitica/blob/9213ee92de43d58c3980d6f89d87c052da627e93/website/server/libs/baseModel.js#L60) and this function, but maybe you have any other ideas?

Reply:
That's something we can revisit in future but for now let's skip this part, in this case the custom toJSON doesn't add anything really necessary